mesonbuild/modules/python.py

```diff
@@ -334,6 +335,28 @@ def __init__(self, name: str, command: T.Optional[T.List[str]] = None,
             'version': '0.0',
         }
 
+    def sanity(self) -> 'PythonExternalProgram':
```

Comment:
this seems wrong, it can return a `NonExistingExternalProgram`, which is a sibling class, right? So it either needs to be `ExternalProgram`, or it needs to be `T.Union[PythonExternalProgram, NonExistingExternalProgram]`

Reply:
True. I slapped this on at the end trying desperately to quell mypy, it's not my area of expertise.